src/create.cpp

```diff
@@ -158,42 +158,44 @@ static void displayCharacterHistory() {
     }
 }
 
-// Clear the previous history strings
-static void playerClearHistory() {
-    for (auto &entry : py.misc.history) {
-        entry[0] = '\0';
+static int16_t calculateSocialClass (int32_t social_class) {
+    if (social_class > 100) {
+        return 100;
+    } else if (social_class < 1) {
+        return 1;
     }
+    return (int16_t) social_class;
 }
 
-// Get the racial history, determines social class -RAK-
-//
-// Assumptions:
-//   - Each race has init history beginning at (race-1)*3+1
-//   - All history parts are in ascending order
-static void characterGetHistory() {
-    int history_id = py.misc.race_id * 3 + 1;
-    int social_class = randomNumber(4);
-
+typedef struct {
+    int32_t social_class;
     char history_block[240];
-    history_block[0] = '\0';
+} BlockOfHistory;
```

Comment:
I guess I understand your reasoning, but I don't think this usage is correct. The `BlockOfHistory` has a `history_block`, aka `block_of_history` field on it, and then it has this `social_class`, which is not a block of history. This does not seem logical.

With my original comment I was thinking that the `social_class` could be the return value. The function would receive the `social_class` and a reference to `history_block`.